packages/pigeon/platform_tests/test_plugin/android/src/main/kotlin/com/example/test_plugin/TestPlugin.kt

```diff
@@ -867,3 +872,57 @@ class TestPluginWithSuffix : HostSmallApi {
     callback(Result.success(Unit))
   }
 }
+
+object SendInts : StreamIntsStreamHandler() {
+  val handler = Handler(Looper.getMainLooper())
+
+  override fun onListen(p0: Any?, sink: PigeonEventSink<Long>) {
+    var count: Long = 0
+    val r: Runnable =
+        object : Runnable {
+          override fun run() {
+            handler.post {
+              sink.success(count)
+              count++
+            }
+            handler.postDelayed(this, 1000)
+            if (count >= 5) {
+              sink.endOfStream()
+            }
+          }
+        }
+    handler.postDelayed(r, 1000)
+  }
+}
+
+object SendClass : StreamEventsStreamHandler() {
+  val handler = Handler(Looper.getMainLooper())
+  val eventList =
+      listOf(
+          IntEvent(1),
+          StringEvent("string"),
+          BoolEvent(false),
+          DoubleEvent(3.14),
+          ObjectsEvent(true),
+          EnumEvent(EventEnum.FORTY_TWO),
+          ClassEvent(EventAllNullableTypes(aNullableInt = 0)))
+
+  override fun onListen(p0: Any?, sink: PigeonEventSink<EventChannelDataBase>) {
+    var count: Int = 0
+    val r: Runnable =
+        object : Runnable {
+          override fun run() {
+            if (count >= eventList.size) {
+              sink.endOfStream()
+            } else {
+              handler.post {
+                sink.success(eventList[count])
+                count++
+              }
+              handler.postDelayed(this, 10)
```

Comment:
Do we actually need a delay here? It would be nice not to add artificial delays to the tests.